app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/location/LocationPickerHelper.kt

```diff
@@ -117,9 +118,18 @@ class LocationPickerHelper @Inject constructor(
         timeoutJob.start()
     }
 
+    @Suppress("TooGenericExceptionCaught")
     internal fun isLocationServicesEnabled(): Boolean {
-        val locationManager = context.getSystemService(Context.LOCATION_SERVICE) as LocationManager
-        return LocationManagerCompat.isLocationEnabled(locationManager)
+        return try {
+            val locationManager = context.getSystemService(Context.LOCATION_SERVICE) as LocationManager
+            LocationManagerCompat.isLocationEnabled(locationManager)
+        } catch (e: Exception) {
+            appLogger.i(
+                message = "Checking for location services availability failed",
+                throwable = e
+            )
+            false
```

Comment:
returning false here will make it seems like location is only disabled when the whole API is missing, maybe returning an Either or an Enum to reflect the state is better for usability, what do you think @m-zagorski

Reply:
If we were to display different message for the end users it would make sense to have some kind of enum or something else here, but as long as we're about to rely on what we have currently (ie the message about location not available) it probably doesnt make sense.